test/e2e/managed_cluster_validating_webhooks_test.go

```diff
@@ -201,42 +217,31 @@ var _ = Describe("Managed Cluster Validating Webhooks", Ordered, func() {
 			}
 		})
 
-		withNamespace := func(pod *v1.Pod, namespace string) *v1.Pod {
-			pod.SetNamespace(namespace)
-			return pod
-		}
-
 		It("blocks pods scheduled onto master/infra nodes", func(ctx context.Context) {
-			err := dedicatedAdmink8s.Create(ctx, withNamespace(pod, privilegedNamespace))
+			err := dedicatedAdmink8s.Create(ctx, pod)
 			Expect(errors.IsForbidden(err)).To(BeTrue())
 
-			err = userk8s.Create(ctx, withNamespace(pod, privilegedNamespace))
+			err = userk8s.Create(ctx, pod)
 			Expect(errors.IsForbidden(err)).To(BeTrue())
 
-			err = userk8s.Create(ctx, withNamespace(pod, unprivilegedNamespace))
+			pod.SetNamespace(unprivilegedNamespace)
+			err = userk8s.Create(ctx, pod)
 			Expect(errors.IsForbidden(err)).To(BeTrue())
+			pod.SetNamespace(privilegedNamespace)
 		}, SpecTimeout(createPodWaitDuration.Seconds()+deletePodWaitDuration.Seconds()))
 
 		It("allows cluster-admin to schedule pods onto master/infra nodes", func(ctx context.Context) {
-			sa := &v1.ServiceAccount{}
-
-			err := client.Get(ctx, saName, namespaceName, sa)
-
-			if err == nil {
-				err = client.Delete(ctx, sa)
-				Expect(err).ToNot(HaveOccurred(), "Failed to delete existing Service Account")
-			}
-
-			sa = &v1.ServiceAccount{
```

Comment:
I found this code to be confusing, as it seems to imply that the `webhook-sa` service name is present in rosa cluster's , and I have not found this to be the case (if I'm wrong, please correct me).

The intent here is to simplify by just having the spec create the `sa` and then also be responsible for removing it.